extensions/machine-learning-services/src/widgets/configTable.ts

```diff
@@ -0,0 +1,182 @@
+							},
+							rowCssStyles: {
+								'border-top': 'solid 1px #ccc',
+								'border-bottom': 'solid 1px #ccc',
+								'border-left': 'none',
+								'border-right': 'none'
+							},
+						},
+						{ // Status icon
+							displayName: constants.mlsConfigStatus,
+							ariaLabel: constants.mlsConfigStatus,
+							valueType: azdata.DeclarativeDataType.component,
+							isReadOnly: true,
+							width: 25,
+							headerCssStyles: {
+								'border': 'none',
+								'background-color': '#FFFFFF'
+							},
+							rowCssStyles: {
+								'border-top': 'solid 1px #ccc',
+								'border-bottom': 'solid 1px #ccc',
+								'border-left': 'none',
+								'border-right': 'none'
+							},
+						},
+						{ // Action
+							displayName: '',
+							valueType: azdata.DeclarativeDataType.component,
+							isReadOnly: true,
+							width: 150,
+							headerCssStyles: {
+								'border': 'none',
+								'background-color': '#FFFFFF',
+								...constants.cssStyles.tableHeader
+							},
+							rowCssStyles: {
+								'border-top': 'solid 1px #ccc',
+								'border-bottom': 'solid 1px #ccc',
+								'border-left': 'none',
+								'border-right': 'none'
+							},
+						}
+					],
+					data: [],
+					ariaLabel: constants.mlsConfigTitle
+				})
+			.component();
+	}
+
+	/**
+	 * Returns the config table component
+	 */
+	public get component(): azdata.DeclarativeTableComponent {
+		return this._statusTable;
+	}
+
+	/**
+	 * Refreshes the config table
+	 */
+	public async refresh(): Promise<void> {
+		this._loadingComponent.updateProperties({ loading: true });
+		let connection = await this.getCurrentConnection();
+		const speesConfig = await this.addTableRow(constants.mlsExternalExecuteScriptTitle,
+			async () => {
+				return await this._serverConfigManager.isMachineLearningServiceEnabled(connection);
+			}, async (enable) => {
+				await this._serverConfigManager.updateExternalScriptConfig(connection, enable);
+				this.refresh();
+			}
+		);
+		const pythonConfig = await this.addTableRow(constants.mlsPythonLanguageTitle,
+			async () => {
+				return await this._serverConfigManager.isPythonInstalled(connection);
+			}, async () => {
+				await this._serverConfigManager.openInstallDocuments();
+			}
+		);
+		const rConfig = await this.addTableRow(constants.mlsRLanguageTitle,
+			async () => {
+				return await this._serverConfigManager.isRInstalled(connection);
+			}, async () => {
+				await this._serverConfigManager.openInstallDocuments();
+			}
+		);
+		this._statusTable.data = [speesConfig, pythonConfig, rConfig];
+		this._loadingComponent.updateProperties({ loading: false });
+	}
+
+	private async getCurrentConnection(): Promise<azdata.connection.ConnectionProfile> {
+		return await this._apiWrapper.getCurrentConnection();
+	}
+
+	private async addTableRow(configName: string, checkEnabledFunction: () => Promise<boolean>, updateFunction: (enable: boolean) => Promise<void>): Promise<azdata.Component[]> {
```

Comment:
>addTableRow [](start = 15, length = 11)

Would suggest changing to "createTableRowComponents" since that's a more accurate description of what it's doing.